### ISSUE TITLE
Updated installed equipment cost formulation

### ISSUE THREAD

**Issue by yoelcortes** [BioSTEAMDevelopmentGroup/biosteam#59]
Previously, the installed equipment cost was calculated as follows:

C_BM  = C_P * F_BM

Which only applies when the bare-module factor is for the given material, pressure, and design. Now we use the more complete formula:

C_BM = C_Pb * (F_BM + F_D * F_P * F_M - 1)

Where:
* C_BM is the bare module cost
* C_Pb is the baseline purchase cost (without accounting for design, pressure, and material),
* F_BM is the bare-module factor
* F_D is the design factor (i.e. type factor)
* F_P is the pressure factor
* F_M is the material factor

All previous code should still work, but if you're working with different design, pressure or material factors, you can use the new `F_D`, `F_P`and `F_M` dictionaries to specify them. You'll also need to add values to the `baseline_purchase_costs` dictionary (not the `purchase_costs` dictionary), then BioSTEAM will take care of calculating both the actual purchase costs (after accounting for construction material and design) and the installed equipment costs.

Here is an example:

```python
import biosteam as bst
class Tank(bst.Unit):
    _BM = {'Tank': 2.3}

    def _cost(self):
        self.F_M['Tank'] = 1.25
        self.baseline_purchase_costs['Tank'] = 1e6

bst.settings.set_thermo(['Water'])
T1 = Tank(ins=bst.Stream(Water=10))
T1.simulate()
print(T1.baseline_purchase_costs)
print(T1.purchase_costs) # Correctly accounts for material factor
print(T1.installed_costs) # Correctly accounts for bare-module and material factor

```
Output:
```
{'Tank': 1000000.0}
{'Tank': 1250000.0}
{'Tank': 2550000.0}
```

Note that adding values to the `baseline_purchase_costs` dictionary is preferable over the `purchase_costs` dictionary, regardless if you are using different material factors or not. You'll get a helpful warning if you add items to the `purchase_costs` dictionary instead of the `baseline_purchase_costs`. For more details, you can checkout the code:

https://github.com/BioSTEAMDevelopmentGroup/biosteam/blob/af1faa9548521cfa3732317bb5dedbc32d0513fc/biosteam/_unit.py#L310

@yalinli2, I also made BioSTEAM spit out a warning if the BM is not included for a purchase cost item (instead of raising an error).

Hope this helps!

**Comment by yalinli2** [comment]
Hi Yoel, thanks for the notice, some questions:
- I'm confused by what ``C_BM`` is. You said:
> Previously, the installed equipment cost was calculated as follows:
>
>        C_BM = C_P * F_BM
But you also said:

> C_BM = C_Pb * (F_BM + F_D * F_P * F_M - 1)
> 
> Where:
> 
> - C_BM is the bare module cost

So is ``C_BM`` installed cost or the bare module cost? Are they the same?
- Also, in the ``_load_capital_costs `` function, can you add the reference to this equation?
- Additionally, is this equation only applies when ``F_D``, ``F_P``, and ``F_M`` are all >1? What if I want to use some cost from stainless steel at higher pressure to calculate cost for carbon steel at lower pressure? Will this result in negative value?

Thanks!

**Comment by yoelcortes** [comment]
Hi Yalin,

Here are the answers to your questions:

* Check out equation 16.12 of Warren's Product and Process Design Principles:
![image](https://user-images.githubusercontent.com/29529649/113511451-6e318600-9525-11eb-960b-ef64dbf262e5.png)

* The bare-module cost and the installed equipment cost are the same.

* F_D, F_P and F_M can all be any number (less or more than 1). What is important is for what material/design/and pressure was the BM applicable. Usually BMs are given for carbon steel at low to moderate pressures. Negative values are impossible as BM is always greater than 1. 

Let me know if you have more questions!

**Comment by yalinli2** [comment]
This all looks good, thanks for explaining and adding the references!

**Comment by yoelcortes** [comment]
Quick update; The factors are now public (so they are called F_D, F_P, and F_M now).

**Comment by yalinli2** [comment]
Cool! Is F_BM public?

**Comment by yoelcortes** [comment]
@yalinli2, yeah, I've been thinking about making the `_BM` dictionary the default values for a public `F_BM` dictionary that gets created with the unit instance. I think this would be better than having bare-module factors as class attributes... which leaves little room for flexibility.

What do you think? Thumbs up and I'll go ahead and fix this up over the weekend.

**Comment by yalinli2** [comment]
I'm on board! 👍

**Comment by yoelcortes** [comment]
Done!

Additionally, to make it easier to understand for future new users, I renamed `_BM` to `_F_BM_default`. I also added `equipment_lifetime` as an instance attribute and renamed the `_equipment_lifetime` class attribute to `_default_equipment_lifetime` (again, to make it easier to understand without necessarily reading documentation).

I added to some checks for Unit subclassing to make sure '_BM' and '_equipment_lifetime' are not used anymore. All tests passed on my computer.

Hopefully it's not too much trouble to make the renames in QSDsan (just import, let it error, and rename).

Thanks!
